boards/arm/particle_xenon/particle_xenon.dts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2018 Endre Karlson
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+	model = "particle xenon";
+	compatible = "particle,particle_xenon", "nordic,nrf52840-qiaa",
+		     "nordic,nrf52840";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,uart-mcumgr = &uart0;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		user_led: led_0 {
+			gpios = <&gpio1 12 0>;
+			label = "User LED";
+		};
+		status_green: led_1 {
+				gpios = <&gpio1 14 0>;
+				label = "Green LED";
+		};
+		status_blue: led_2 {
+				gpios = <&gpio1 15 0>;
+				label = "Blue LED";
+		};
+		status_red: led_3 {
+				gpios = <&gpio1 13 0>;
+				label = "Red LED";
+		};
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		mode_button: button_0 {
+			gpios = <&gpio0 11 GPIO_PUD_PULL_UP>;
+			label = "Mode Button";
+		};
+	};
+
+	aliases {
+		led0 = &user_led;
+		led1 = &status_green;
+		led2 = &status_blue;
+		led3 = &status_red;
+		sw0 = &mode_button;
+	};
+};
+
+&gpiote {
+	status ="ok";
+};
+
+&gpio0 {
+	status ="ok";
+};
+
+&gpio1 {
+	status ="ok";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "ok";
+	tx-pin = <6>;
+	rx-pin = <8>;
+};
```

Comment:
Add rts-pin and cts-pin, as with pca10056.  (I now understand how the magic works for GPIOs specified as integers: they get passed through directly to the nrfcx hal which knows how to convert the linearized ones to instance-specific psel values.)